### ISSUE TITLE
Switching to Player View with Object flagged Visible over Fog of War can cause an NPE

### ISSUE THREAD

**Issue by Phergus** [RPTools/maptool#2313]
**Describe the bug**
Under specific conditions switching to Player View on a map with one or more objects that are set to `Visible over Fog of War` can produce an NPE.

```
java.lang.NullPointerException: Cannot invoke "java.awt.geom.Area.intersects(java.awt.geom.Rectangle2D)" because "fog" is null
	at net.rptools.maptool.model.Grid.checkRegion(Grid.java:688)
	at net.rptools.maptool.client.ui.zone.ZoneRenderer.isTokenInNeedOfClipping(ZoneRenderer.java:3755)
	at net.rptools.maptool.client.ui.zone.ZoneRenderer.renderTokens(ZoneRenderer.java:3172)
	at net.rptools.maptool.client.ui.zone.ZoneRenderer.renderZone(ZoneRenderer.java:1362)
	at net.rptools.maptool.client.ui.zone.ZoneRenderer.paintComponent(ZoneRenderer.java:885)
	at java.desktop/javax.swing.JComponent.paint(Unknown Source)
...
```
- Map needs to have Fog of War turned on. 
- Map needs Vision set to Day (night may also do it).
- Have a token on Object layer with Visible over FoW enabled.

**To Reproduce**
1. Load attached campaign file.
2. Either from View menu or via Ctrl-Shift-P switch to Player View.
3. See NPE.

**Expected behavior**
Player view should work correctly.

**Workaround:**  Adding a token with vision to the map prevents the NPE.

**MapTool Info**
- Version: 1.8.0-beta.2
- Install: New install

**Desktop (please complete the following information):**
 - OS: Windows
 - Version: 10

**Additional context**
Reported by Rumble on Discord
https://discordapp.com/channels/296230822262865920/296230822262865920/780086848533168168

NPE does not occur with current dev code.

Test Campaign:  
[NPE VBL.cmpgn.zip](https://github.com/RPTools/maptool/files/5580081/NPE.VBL.cmpgn.zip)
(Remove `.zip` from filename.)

**Comment by Phergus** [comment]
Even though this is apparently fixed in current dev code, opening issue so that others encountering the bug can perhaps find info here.

**Comment by FullBleed** [comment]
I know it's tangential to this bug, but I posted a bug on the forums related to states on Objects with Visible over Fog of War being visible through fog when the object isn't.  Here is a screenshot where a "locked" state is applied to a door that is on the object layer.  The door is set to be visible over fow and only shows when it is within a token's vision range.  The locked state, however, is visible all the time... floating out in space.  This is not the case when the door is on the token layer. 
![States on Objects with Visible over FoW](https://user-images.githubusercontent.com/37131742/100054485-1f921380-2df0-11eb-98e0-1a6980546fb1.jpg)

**Comment by Phergus** [comment]
Was fixed by #2243 .

**Comment by FullBleed** [comment]
Great!  Lately I've been trying to use objects for, you know, "objects" (doors, keys, chests, etc.) and I've been finding that I don't think many people do given their current state (i.e. this bug, not being able to "select" them and run macros on them easily without switching to their layer or building getToken area macros, having a significantly diminished context menu compared to tokens, etc.)  This removes one hurdle in my current usage though so just one more reason to be looking forward to 1.8.  Those floating lock states on my maps kinda give some things away.  ;)
